### ISSUE TITLE
Add Super2xSal support

### ISSUE THREAD

**Issue by RefriZaddo** [AppleWin/AppleWin#1136]
just that

**Comment by tomcw** [comment]
Here's some high-level info: https://en.wikipedia.org/wiki/Pixel-art_scaling_algorithms#2%C3%97SaI

And the original 2xSaI and Super2xSaI: https://vdnoort.home.xs4all.nl/emulation/2xsai/
(With source code & 2XSAI.TXT [here](https://vdnoort.home.xs4all.nl/emulation/2xsai/2xSaI_sources.rar))
The source code is GPL'ed, so let me just add it here: [2xSaI_sources.zip](https://github.com/AppleWin/AppleWin/files/9794339/2xSaI_sources.zip)


Thanks for your suggestion.

From the 2XSAI.TXT:

>How big are the scaled images?
==============================
2xSaI, SuperEagle and Super2xSaI scale the dimensions of the image
by a factor of 2 [area is 2*2 = 4 times larger]. Scale2xSaI scales the
image to an arbitrary LARGER size. Smaller also works, but will look like
crap in most situations since it was not written for that purpose.

NB, it's the Scale2xSaI algorithm that does arbitrary scaling.

This code is >20 years old, and uses the CPU to do bilinear interpolation for 2x (or arbitrary) upscaling.
This [GH project](https://github.com/VladislavPVI/Bilinear-interpolation-CUDA) (from 2020) shows CPU vs GPU timings, with GPU being 2-3.6 times faster than the CPU.

State of the art is using AI algorithms for super-resolution, eg. Nvidia's DLSS & AMD's [FSR](https://www.tomshardware.com/reference/amd-fsr-fidelityfx-super-resolution-explained) and FSR2.

**Comment by audetto** [comment]
Very interesting.

Since AW runs at < 1% CPU, I think we can live with a CPU implementation (to begin with), as long as it takes a number of milliseconds which fits in a hand.

**Comment by audetto** [comment]
The source code is small, but it looks so ancient.

It probably comes from a 16 bit world with all the issues around accessing memory in different segments.

http://www.delorie.com/djgpp/doc/libc/libc_304.html

It probably deserves anyway to be ported to a 32/64 world where none of these issues exist any longer.

**Comment by tomcw** [comment]
>Since AW runs at < 1% CPU, I think we can live with a CPU implementation (to begin with)

Yes, I agree - it should be straightforward (although I've not inspected the SaI code).

btw, there are 3 occurrences of `StretchBlt()` - but it's only the one in `void Win32Frame::VideoPresentScreen(void)` which would need changing.  (The other 2 are for: the AppleWin logo at start-up, and the debugger, which don't need bilinear upscaling.)

**Comment by RefriZaddo** [comment]
i'm glad to see a talk in my issue
